e2e/gatsby/src/gatsby.test.ts

```diff
@@ -31,30 +32,35 @@ describe('Gatsby Applications', () => {
       return updated;
     });
 
-    let result = runCLI(`build ${appName}`);
-    expect(result).toContain('Done building in');
+    runCLI(`build ${appName}`);
+    checkFilesExist(
+      `apps/${appName}/public/index.html`,
+      `apps/${appName}/public/404.html`,
+      `apps/${appName}/public/manifest.webmanifest`
+    );
 
-    result = runCLI(`lint ${appName}`);
+    const result = runCLI(`lint ${appName}`);
     expect(result).not.toMatch('Lint errors found in the listed files');
 
-    const testResults = await runCLIAsync(`test ${appName}`);
-    expect(testResults.combinedOutput).toContain(
-      'Test Suites: 2 passed, 2 total'
-    );
+    expectTestsPass(await runCLIAsync(`test ${appName}`));
   }, 600000);
 
   it('should support styled-jsx', async () => {
     const appName = uniq('app');
 
     runCLI(`generate @nrwl/gatsby:app ${appName} --style styled-jsx`);
 
-    let result = runCLI(`build ${appName}`);
-    expect(result).toContain('Done building in');
+    runCLI(`build ${appName}`);
+    checkFilesExist(
+      `apps/${appName}/public/index.html`,
+      `apps/${appName}/public/404.html`,
+      `apps/${appName}/public/manifest.webmanifest`
+    );
```

Comment:
got a failed test locally despite the log has 'Done building in', so i just change to check whether files exist:
```
info Done building in 27.271117298 sec··
    Pages·
    ┌ src/pages/404.js
    │ ├   /404/
    │ └   /404.html
    └ src/pages/index.tsx
      └   /·
      ╭────────────────────────────────────────────────────────────────╮
      │                                                                │
      │   (SSG) Generated at build time                                │
      │ D (DSG) Deferred static generation - page generated at runtime │
      │ ∞ (SSR) Server-side renders at runtime (uses getServerData)    │
      │ λ (Function) Gatsby function                                   │
      │                                                                │
      ╰────────────────────────────────────────────────────────────────╯··
    ———————————————————————————————————————————————·
    >  NX   SUCCESS  Running target \"build\" succeeded··
    "}

      34 |
      35 |     let result = await runCLIAsync(`build ${appName}`);
    > 36 |     expect(result).toContain('Done building in');
```

Reply:
Yup makes sense... the output for CLI can be truncated or have `\n` inserted at unexpected points.